tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AWS/SDK/HttpWebRequestMessageGetResponseIntegration.cs

```diff
@@ -0,0 +1,45 @@
+// <copyright file="HttpWebRequestMessageGetResponseIntegration.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+#nullable enable
+
+using System;
+using System.ComponentModel;
+using Datadog.Trace.ClrProfiler.CallTarget;
+using Datadog.Trace.DuckTyping;
+using Datadog.Trace.Tagging;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.AWS.SDK;
+
+/// <summary>
+/// Amazon.Runtime.Internal.Transform.IWebResponseData Amazon.Runtime.HttpWebRequestMessage::GetResponse() calltarget instrumentation
+/// </summary>
+[InstrumentMethod(
+    AssemblyName = "AWSSDK.Core",
+    TypeName = "Amazon.Runtime.HttpWebRequestMessage",
+    MethodName = "ProcessHttpResponseMessage",
+    ReturnTypeName = "Amazon.Runtime.Internal.Transform.HttpClientResponseData",
+    ParameterTypeNames = [ClrNames.HttpResponseMessage],
+    MinimumVersion = "4.0.0",
+    MaximumVersion = "4.*.*",
+    IntegrationName = AwsConstants.IntegrationName)]
+[Browsable(false)]
+[EditorBrowsable(EditorBrowsableState.Never)]
+public class HttpWebRequestMessageGetResponseIntegration
+{
+    internal static CallTargetState OnMethodBegin<TTarget, TResponseMessage>(TTarget instance, TResponseMessage? responseMessage)
+        where TTarget : IHttpWebRequestMessage, IDuckType
+    {
+        // V4 of the SDK uses a templating system to generate the request Uri and other data
```

Comment:
This is the only "new" part of the instrumentation that I had to add, to ensure the S3 spans have the correct outgoing URLs